docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialFullConvolution.md

```diff
@@ -0,0 +1,250 @@
+```
+
+SpatialFullConvolution is a module that applies a 2D full convolution over an input image. 
+
+The input tensor in `forward(input)` is expected to be
+either a 4D tensor (`batch x nInputPlane x height x width`) or a 3D tensor (`nInputPlane x height x width`). The convolution is performed on the last two dimensions. `adjW` and `adjH` are used to adjust the size of the output image. The size of output tensor of `forward` will be :
+```
+  output width  = (width  - 1) * dW - 2*padW + kW + adjW
+  output height = (height - 1) * dH - 2*padH + kH + adjH
+``` 
+
+Note, scala API also accepts a table input with two tensors: `T(convInput, sizeTensor)` where `convInput` is the standard input tensor, and the size of `sizeTensor` is used to set the size of the output (will ignore the `adjW` and `adjH` values used to construct the module). Use `SpatialFullConvolution[Table, T](...)` instead of `SpatialFullConvolution[Tensor,T](...)`) for table input.
+ 
+This module can also be used without a bias by setting parameter `noBias = true` while constructing the module.
+ 
+Other frameworks may call this operation "In-network Upsampling", "Fractionally-strided convolution", "Backwards Convolution," "Deconvolution", or "Upconvolution."
+ 
+Reference: Long J, Shelhamer E, Darrell T. Fully convolutional networks for semantic segmentation[C]//Proceedings of the IEEE Conference on Computer Vision and Pattern Recognition. 2015: 3431-3440.
+
+Detailed explaination of arguments in constructor. 
+
+ * param nInputPlane The number of expected input planes in the image given into forward()
+ * param nOutputPlane The number of output planes the convolution layer will produce.
+ * param kW The kernel width of the convolution.
+ * param kH The kernel height of the convolution.
+ * param dW The step of the convolution in the width dimension. Default is 1.
+ * param dH The step of the convolution in the height dimension. Default is 1.
+ * param padW The additional zeros added per width to the input planes. Default is 0.
+ * param padH The additional zeros added per height to the input planes. Default is 0.
+ * param adjW Extra width to add to the output image. Default is 0.
+ * param adjH Extra height to add to the output image. Default is 0.
+ * param nGroup Kernel group number.
+ * param noBias If bias is needed.
+ * param wRegularizer: instance of [[Regularizer]]
+ *                    (eg. L1 or L2 regularization), applied to the input weights matrices.
+ * param bRegularizer: instance of [[Regularizer]]
+ *                    applied to the bias.
+ 
+**Scala example:**
+
+Tensor Input example: 
+
+```scala
+
+scala>
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor._
+import com.intel.analytics.bigdl.tensor.Storage
+
+val m = SpatialFullConvolution[Tensor[Float],Float](1, 2, 2, 2, 1, 1,0, 0, 0, 0, 1, false)
```

Comment:
I just merge the doc as is now, to keep in sync with code. Later when the layer is fixed, doc should be changed together in the same PR.